cloudinit/sources/DataSourceEc2.py

```diff
@@ -731,28 +735,97 @@ def convert_ec2_metadata_network_config(network_md, macs_to_nics=None,
        not provided, get_interfaces_by_mac is called to get it from the OS.
     @param: fallback_nic: Optionally provide the primary nic interface name.
        This nic will be guaranteed to minimally have a dhcp4 configuration.
+    @param: apply_network_config: Boolean set True to configure all networking
```

Comment:
renamed to full_network_config for this function. I'm not sure if you also apply this statement to the datasource config option which I still left named `apply_network_config` because both OpenStack and  Azure have the same config option and I was trying to align with the other existing public config options.